### ISSUE TITLE
Feature Request: Allow capturing of alternative text

### ISSUE THREAD

**Issue by ddaypunk06** [cucumber/cucumber-jvm#1673]
## Summary

It would be nice if Cucumber Expressions could capture the alternative text from the feature file and use it as a parameter as we were able to with regex:

Ex.
`@And("^I set the (current|comparison) date range to \"([^\"]*)\" through \"([^\"]*)\"$")`

vs.

`@And("I set the (current/comparison) date range to {string} through {string}")`

## Expected Behavior

Cucumber users are able to capture alternate text as a parameter to the step definition method.

## Current Behavior

Unable to do this currently

```


```
## Possible Solution

Allow for something like `(choice1/choice2)` which would capture that word in that position without quotes around it and pass it as an argument.

## Context & Motivation

Updating from cuke 1 to cuke 4 (our stuff is old) and I hit a roadblock here. My current solution is to make an enum and pass that down instead, but this could be nice for more than just my team.

**Comment by mpkorstanje** [comment]
While somewhat counter intuitive at first best practice here is to create two different step definitions. And factor out the common code to a helper method.

Trying to capture the alternative is an admission that this step is not doing one thing but actually a combination of two steps that do different things.

You'll also find that while the extra step adds some code the code it self is uncomplicated. Where as now you are reducing code somewhat at the cost of needless complexity. When writing tests low complexity is generally preferable. It makes it easier to validate their correctness.

**Comment by mpkorstanje** [comment]
Example, rather sparse because of phone:

```
@And("I set the current date range to {string} through {string}")
...
    setDateRange(CURRENT, from, to);
....

@And("I set the comparison date range to {string} through {string}")
...
    setDateRange(COMPARISON, from, to);
...
```

**Comment by ddaypunk06** [comment]
@mpkorstanje I don't see:
`@And("I set the (current|comparison) date range to {string} through {string}")
...
    setDateRange(range, from, to);
...`

...as all that much more complicated honestly. The code the step is running is still making a choice lower down. I've actually made my tests much easier to maintain by combining similar steps across our app using the group capture pattern with regex. I'd rather have one step calling the code than possibly two steps in different packages calling that piece of code.

Also, as Cucumber is a tool, I think it should be up to the user to decide best practice on how to write their steps. I also don't see said best practice stated somewhere, unless I missed it? Unless you are referring to OOP. Regardless, I don't see either way as more complex or clean than the other, but I see more utility in my suggestion for consumers of the tool.
